packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -500,12 +500,15 @@ class Route extends EmberObject implements IRoute {
     export default class ApplicationRoute extends Route {
       @action
       loading(transition, route) {
-        let controller = this.controllerFor('foo');
-        controller.set('currentlyLoading', true);
+        const controller = this.controllerFor('foo');
 
-        transition.finally(function() {
-          controller.set('currentlyLoading', false);
-        });
+        if (controller) {
```

Comment:
@nickeidler makes sense that controllers aren't always instantiated now that I stop to think about it, but it doesn't seem immediately apparent. Might be worth a code comment above this line explaining that briefly. Something like:

```suggestion
        // controller may not be instantiated when initially loading
        if (controller) {
```

Reply:
Good point-- added a comment.